app/controllers/rooms_controller.rb

```diff
@@ -138,6 +141,27 @@ def destroy
     redirect_to current_user.main_room
   end
 
+  # POST room/join
+  def join_specific_room
+    parsed_url = params[:join_room][:url].split('/')
+
+    room_uid = parsed_url[parsed_url.length - 1]
+
+    begin
+      @room = Room.find_by(uid: room_uid)
+    rescue ActiveRecord::RecordNotFound
+      flash[:alert] = I18n.t("room.no_room.invalid_room_uid")
+      return redirect_to room_path(current_user.main_room)
```

Comment:
You should be able to just `redirect_to current_user.main_room` with the flash in one line